tests/networkpolicy_test.go

```diff
@@ -0,0 +1,246 @@
+package tests_test
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/google/goexpect"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1network "k8s.io/api/networking/v1"
+	v13 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/tests"
+)
+
+var _ = Describe("Networkpolicy", func() {
+
+	flag.Parse()
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	tests.PanicOnError(err)
+
+	var vmia *v1.VirtualMachineInstance
+	var vmib *v1.VirtualMachineInstance
+	var vmic *v1.VirtualMachineInstance
+
+	tests.BeforeAll(func() {
+		tests.SkipIfUseFlannel(virtClient)
+		tests.BeforeTestCleanup()
+		// Create three vmis, vmia and vmib are in same namespace, vmic is in different namespace
+		vmia = tests.NewRandomVMIWithEphemeralDiskAndUserdataAndNS(tests.RegistryDiskFor(tests.RegistryDiskCirros), tests.NamespaceTestDefault, "#!/bin/bash\necho 'hello'\n")
+		vmia.Labels = map[string]string{"type": "test"}
+		vmia, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmia)
+		Expect(err).ToNot(HaveOccurred())
+		tests.WaitForSuccessfulVMIStart(vmia)
+
+		vmib = tests.NewRandomVMIWithEphemeralDiskAndUserdataAndNS(tests.RegistryDiskFor(tests.RegistryDiskCirros), tests.NamespaceTestDefault, "#!/bin/bash\necho 'hello'\n")
```

Comment:
I think this sort of construct more correctly belongs in a "BeforeEach" clause. The "BeforeAll" utility function is designed to run once before all unit tests. Doing it once at the beginning (vs once for every test) means the VMI's and other test fixtures being set up here are not being reset between uses--so any side effects from one test will break unrelated tests at random.

Reply:
Actually I thought if I put this in "BeforeAll", then we can use these vmis in the whole networkpolicy test. If put this in "BeforeEach", seems we need to create/delete vmis for every context. But if it is better to put it in "BeforeEach", I could update it. Thanks:)

Reply:
but the test does not make any modifications to the VMIs  - only the network policy objects. seems to me like a legitimate optimization f the execution time.

Reply:
Yes, that was my intention.